arch/arm/src/samv7/sam_dac.c

```diff
@@ -226,15 +226,46 @@ static int dac_interrupt(int irq, void *context, void *arg)
 
 static void dac_reset(struct dac_dev_s *dev)
 {
+  struct sam_chan_s *chan = dev->ad_priv;
+#ifdef CONFIG_SAMV7_DAC_TRIGGER
+  uint32_t regval;
+#endif
   irqstate_t flags;
 
   /* Reset only the selected DAC channel; the other DAC channel must remain
-   * functional.
+   * functional. The controller however does not have an option to reset
+   * single channel, therefore we have to do this manually by writing zeroes
+   * to all important registers.
+   */
+
+  /* This should be called only before dac_setup(), therefore the channel
+   * should not be in use. Skip reset if it is.
    */
 
-  flags   = enter_critical_section();
+  if (chan->inuse)
```

Comment:
maybe convert `inuse` to `bool`? Anyway current implementation does not bring any benefit unless we change `intf` to be a bitfield

Reply:
I have changed it to `bool`, it is probably clearier. Bud what do you mean by changing `intf` to bitfield, channel enable and channel disable are both write only, so we just need to write `1 << offset`, where offset is interface number.

Reply:
I've read a comment around `intf` and it states that it can be either 0 or 1, so in theory it could be defined as `uint8_t intf : 1;` (if it had to care more values obviously `uint8_t intf : 7;` also could work) so it would share memory with `inuse` bit field.

Reply:
@pkarashchenko Oh yes, i thought you were suggesting `0b01` value for channel 0 and `0b10` for channel 1. In that case yes.